goldstone/core/models.py

```diff
@@ -2657,3 +2657,94 @@ def _get_traits_dict(e):
         :return: dict
         """
         return {"traits": e.as_dict()}
+
+
+class AlertSearch(SavedSearch):
+    """
+        Model for AlertSearches, a subclass of SavedSearch that has an alert
+        notification configured to be sent out to one or more recipients.
+
+    """
+
+    class Meta:               # pylint: disable=C0111,W0232,C1001
+
+        verbose_name_plural = "saved searches with alerts"
+
+    def build_alert_template(self, hits):
+        msg_title_template = 'Alert : ' + str(self.name) + \
+                             ' triggered with ' + str(hits) +\
+                             ' hits in the last ' + str(self.target_interval) +\
+                             ' minutes.'
+
+        msg_body_template = 'There were ' + str(hits) + \
+                            ' occurrences of ' + str(self.name) +\
+                            ' in the last ' + str(self.target_interval) +\
+                            ' minutes.' + '\n' +\
+                            'This query was last updated at : ' +\
+                            str(self.updated) + '\n' +\
+                            'This query last ran from : ' +\
+                            str(self.last_start) + ' to : ' +\
+                            str(self.last_end)
+        msg_params_dict = {'title': msg_title_template,
+                           'body': msg_body_template}
+        return msg_params_dict
+
+
+class Alert(models.Model):
+    """
+        Create an alert instance object. This contains the fleshed-out
+        alert message which will be passed down to the producer interface
+        to be sent out.
+
+        This does not contain msg-params for actually sending out this alert.
+        For those details, refer to class Producer.
+    """
+
+    query = models.ForeignKey(AlertSearch)
+    name = models.CharField(max_length=64, default='generic-alert',
```

Comment:
we already have the alertsearch that triggered it, and the msg_title.  does the name provide any additional value?